de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlShipperGatewayClient.java

```diff
@@ -0,0 +1,420 @@
+import de.dhl.webservices.businesscustomershipping._3.ReceiverType;
+import de.dhl.webservices.businesscustomershipping._3.ShipmentDetailsTypeType;
+import de.dhl.webservices.businesscustomershipping._3.ShipmentItemType;
+import de.dhl.webservices.businesscustomershipping._3.ShipmentNotificationType;
+import de.dhl.webservices.businesscustomershipping._3.ShipmentOrderType;
+import de.dhl.webservices.businesscustomershipping._3.ShipperType;
+import de.dhl.webservices.businesscustomershipping._3.Version;
+import de.metas.shipper.gateway.dhl.model.DhlClientConfig;
+import de.metas.shipper.gateway.dhl.model.DhlCustomDeliveryData;
+import de.metas.shipper.gateway.dhl.model.DhlPackageLabelType;
+import de.metas.shipper.gateway.spi.ShipperGatewayClient;
+import de.metas.shipper.gateway.spi.exceptions.ShipperGatewayException;
+import de.metas.shipper.gateway.spi.model.Address;
+import de.metas.shipper.gateway.spi.model.ContactPerson;
+import de.metas.shipper.gateway.spi.model.DeliveryOrder;
+import de.metas.shipper.gateway.spi.model.DeliveryPosition;
+import de.metas.shipper.gateway.spi.model.PackageDimensions;
+import de.metas.shipper.gateway.spi.model.PackageLabel;
+import de.metas.shipper.gateway.spi.model.PackageLabels;
+import de.metas.shipper.gateway.spi.model.PickupDate;
+import lombok.Builder;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.oxm.jaxb.Jaxb2Marshaller;
+import org.springframework.ws.WebServiceMessage;
+import org.springframework.ws.client.core.WebServiceMessageCallback;
+import org.springframework.ws.client.core.WebServiceTemplate;
+import org.springframework.ws.soap.SoapHeader;
+import org.springframework.ws.soap.SoapMessage;
+import org.springframework.ws.transport.http.HttpComponentsMessageSender;
+import org.springframework.xml.transform.StringResult;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.format.DateTimeFormatter;
+import java.util.Base64;
+import java.util.List;
+
+public class DhlShipperGatewayClient implements ShipperGatewayClient
+{
+	private static final Logger logger = LoggerFactory.getLogger(DhlShipperGatewayClient.class);
+
+	private final Version API_VERSION;
+
+	private final DhlClientConfig config;
+
+	private final WebServiceTemplate webServiceTemplate;
+	private final SoapHeaderWithAuth soapHeaderWithAuth;
+
+	private final de.dhl.webservice.cisbase.ObjectFactory objectFactoryCis;
+	private final de.dhl.webservices.businesscustomershipping._3.ObjectFactory objectFactory;
+
+	@Builder
+	public DhlShipperGatewayClient(@NonNull final DhlClientConfig config)
```

Comment:
*if* it's just the config you need in this client (which would be very nice), i suggest that you use a static build method instead of the lombok-builder.